### PR TITLE
Unit test the startAt filtering

### PR DESCRIPTION
**Pull request by paraita**

### CODE REVIEW

**Comment by paraita** [comment]
retest this please

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 6 issues:
* CRITICAL 1 critical
* MAJOR 4 major
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Reduce this anonymous class number of lines from 33 to at most 20, or make it a named class.](https://github.com/ow2-proactive/scheduling/blob/0461d9166fd4eed26ec027e3a4f0c7541e728b1b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java#L308) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1188)
* MAJOR [Remove this unused "jfc" local variable.](https://github.com/ow2-proactive/scheduling/blob/0461d9166fd4eed26ec027e3a4f0c7541e728b1b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java#L205) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1481)
* MINOR [Remove this unused import 'org.ow2.proactive.scheduler.common.TaskSortParameter'.](https://github.com/ow2-proactive/scheduling/blob/0461d9166fd4eed26ec027e3a4f0c7541e728b1b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java#L41) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)
* MAJOR [Method has 9 parameters, which is greater than 7 authorized.](https://github.com/ow2-proactive/scheduling/blob/0461d9166fd4eed26ec027e3a4f0c7541e728b1b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java#L287) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)
* CRITICAL [Use Hibernate's parameter binding instead of concatenation.](https://github.com/ow2-proactive/scheduling/blob/0461d9166fd4eed26ec027e3a4f0c7541e728b1b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java#L1208) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2077)
* MAJOR [Method has 9 parameters, which is greater than 7 authorized.](https://github.com/ow2-proactive/scheduling/blob/0461d9166fd4eed26ec027e3a4f0c7541e728b1b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java#L274) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)

**Comment by paraita** [comment]
retest this please

**Comment by paraita** [comment]
The sonarqube comment "Use Hibernate's parameter binding instead of concatenation." is not valid.

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 6 issues:
* CRITICAL 1 critical
* MAJOR 4 major
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Remove this unused "jfc" local variable.](https://github.com/ow2-proactive/scheduling/blob/0461d9166fd4eed26ec027e3a4f0c7541e728b1b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java#L205) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1481)
* CRITICAL [Use Hibernate's parameter binding instead of concatenation.](https://github.com/ow2-proactive/scheduling/blob/0461d9166fd4eed26ec027e3a4f0c7541e728b1b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java#L1208) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2077)
* MINOR [Remove this unused import 'org.ow2.proactive.scheduler.common.TaskSortParameter'.](https://github.com/ow2-proactive/scheduling/blob/0461d9166fd4eed26ec027e3a4f0c7541e728b1b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java#L41) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)
* MAJOR [Method has 9 parameters, which is greater than 7 authorized.](https://github.com/ow2-proactive/scheduling/blob/0461d9166fd4eed26ec027e3a4f0c7541e728b1b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java#L287) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)
* MAJOR [Reduce this anonymous class number of lines from 33 to at most 20, or make it a named class.](https://github.com/ow2-proactive/scheduling/blob/0461d9166fd4eed26ec027e3a4f0c7541e728b1b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java#L308) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1188)
* MAJOR [Method has 9 parameters, which is greater than 7 authorized.](https://github.com/ow2-proactive/scheduling/blob/0461d9166fd4eed26ec027e3a4f0c7541e728b1b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java#L274) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)

**Comment by paraita** [comment]
@ow2-proactive/developers can I have two reviews please ?

File: scheduler/scheduler-server/src/test/java/org/ow2/proactive/scheduler/core/db/schedulerdb/TaskDBUtilsTest.java [hunk 1]
```diff
@@ -120,6 +124,11 @@ public void setUp() throws Exception {
             dbManager.updateFinishedTime(
                     job.getId().longValue(), task.getId().longValue(),endTime);
 
+            if (i == NTH_JOB_TO_BE_CROND) {
```
Comment (marcocast) [inline scheduler/scheduler-server/src/test/java/org/ow2/proactive/scheduler/core/db/schedulerdb/TaskDBUtilsTest.java]: could you change the i variable in something better?

**Comment by marcocast** [comment]
:+1:

File: scheduler/scheduler-server/src/test/java/org/ow2/proactive/scheduler/core/db/schedulerdb/TaskDBUtilsTest.java [hunk 2]
```diff
@@ -73,10 +73,14 @@
 
     private static final int NB_JOBS_AS_DEMO_USER = NB_JOBS / 2; // must be <= NB_JOBS
 
+    private static final int NTH_JOB_TO_BE_CROND = 2;
```
Comment (lpellegr) [inline scheduler/scheduler-server/src/test/java/org/ow2/proactive/scheduler/core/db/schedulerdb/TaskDBUtilsTest.java]: What is the meaning of _NTH_?

**Comment by lpellegr** [comment]
The two commits contained by this patch could probably be squashed since they have the same description.

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 6 issues:
* CRITICAL 1 critical
* MAJOR 4 major
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Reduce this anonymous class number of lines from 33 to at most 20, or make it a named class.](https://github.com/ow2-proactive/scheduling/blob/8a0bd9c0a99e5cd3c97f7f0bedf3382bdc570791/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java#L308) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1188)
* CRITICAL [Use Hibernate's parameter binding instead of concatenation.](https://github.com/ow2-proactive/scheduling/blob/8a0bd9c0a99e5cd3c97f7f0bedf3382bdc570791/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java#L1208) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2077)
* MAJOR [Method has 9 parameters, which is greater than 7 authorized.](https://github.com/ow2-proactive/scheduling/blob/8a0bd9c0a99e5cd3c97f7f0bedf3382bdc570791/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java#L287) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)
* MINOR [Remove this unused import 'org.ow2.proactive.scheduler.common.TaskSortParameter'.](https://github.com/ow2-proactive/scheduling/blob/8a0bd9c0a99e5cd3c97f7f0bedf3382bdc570791/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java#L41) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)
* MAJOR [Remove this unused "jfc" local variable.](https://github.com/ow2-proactive/scheduling/blob/8a0bd9c0a99e5cd3c97f7f0bedf3382bdc570791/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java#L205) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1481)
* MAJOR [Method has 9 parameters, which is greater than 7 authorized.](https://github.com/ow2-proactive/scheduling/blob/8a0bd9c0a99e5cd3c97f7f0bedf3382bdc570791/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java#L274) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)

**Comment by paraita** [comment]
@lpellegr the 2 commits have been squashed thanks for the reviews !

**Comment by lpellegr** [comment]
:+1:

**Pull request merged**